spec/requests/distance_request_spec.rb

```diff
@@ -5,24 +5,62 @@
 require './app/controllers/weather_service_app'
 
 
-describe 'Climate Data API' do
+describe 'Distance API' do
   include Rack::Test::Methods
+  describe 'happy path' do
+    it 'has a successful response' do
+      VCR.use_cassette('denver_to_seattle') do
+        params = {
+          origin: 'Denver,CO',
+          destination: 'Seattle,WA'
+        }
+        get "/api/v1/distance?origin=#{params[:origin]}&destination=#{params[:destination]}"
 
-  it 'has a successful response' do
-    VCR.use_cassette('denver_to_seattle') do
+        expect(last_response).to be_ok
+
+        parsed = JSON.parse(last_response.body, symbolize_names: true)
+        data = parsed[:data]
+
+        expect(data).to be_a(Hash)
+
+      end
+    end
+  end
+
+  describe 'sad path' do
+    it 'returns errors for missing params' do
       params = {
         origin: 'Denver,CO',
         destination: 'Seattle,WA'
       }
-      get "/api/v1/distance?origin=#{params[:origin]}&destination=#{params[:destination]}"
+      get "/api/v1/distance?origin=#{params[:origin]}"
+      expect(last_response.status).to eq(401)
+      error = JSON.parse(last_response.body, symbolize_names: true)
 
-      expect(last_response).to be_ok
+      expect(error).to be_a(String)
+      expect(error).to eq('Parameter(s) missing: Please provide a valid origin and destination')
+    end
+
+    it 'returns errors for invalid location' do
+      params = {
+        origin: 'doifhdsfhdjsfl',
```

Comment:
This is great sad path coverage! As I mentioned to Alexa, and this was just coming from my experience with my Sweater Weather eval.. a great edge case we can test for would be if the origin and destination are something like `[]` or `&&&&&` or some real odd characters outsides of just letters. But this is definitely a great start to the sad path testing!